changelog.d/15889.bugfix

```diff
@@ -0,0 +1 @@
+Remove outlier check for getting membership events through the `/_matrix/client/r0/rooms/{roomId}/event/{eventId}` endpoint.
```

Comment:
We should probably create a spec issue to track clarifying the out-of-band membership behavior, https://github.com/matrix-org/matrix-spec/issues